src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs

```diff
@@ -712,18 +724,21 @@ public LoadableClassInfo GetLoadableClassInfo(string loadName, Type signatureTyp
         /// <summary>
         /// Get all registered entry points.
         /// </summary>
-        public IEnumerable<EntryPointInfo> AllEntryPoints()
+        [BestFriend]
+        internal IEnumerable<EntryPointInfo> AllEntryPoints()
         {
             return _entryPoints.AsEnumerable();
         }
 
-        public bool TryFindEntryPoint(string name, out EntryPointInfo entryPoint)
+        [BestFriend]
+        internal bool TryFindEntryPoint(string name, out EntryPointInfo entryPoint)
```

Comment:
same as above, they could prove useful a public.

Reply:
AFAIK, the plan around `ComponentCatalog` is that the *only* publicly accessible member on it will be the thing to register assemblies, as well as whatever we wind up doing to register for custom transformers, right?

I think if we go function by function by what "might" prove useful we'll identify everything as potentially useful -- after all, everything was written for a reason. I think though we might want to be a bit more picky about those scenarios we're actually trying to hit for v1.